src/v/cluster/partition_manager.cc

```diff
@@ -162,10 +163,26 @@ ss::future<> partition_manager::stop_partitions() {
     co_await _gate.close();
     // prevent partitions from being accessed
     auto partitions = std::exchange(_ntp_table, {});
-    _raft_table.clear();
+
+    {
```

Comment:
The idea here is to break up the destructor by basically putting a yield point between destroying every element, right?

We use this 2x here already and maybe elsewhere, could be worth wrapping it up in a utility function?